nonos/geometry.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+
+
+def cartesian_to_cylindrical(X, Y, Z):
+    R = np.sqrt(X ** 2 + Y ** 2)
+    phi = np.arctan2(Y, X)
+    z = Z
+    return (R, phi, z)
+
+
+def cartesian_to_spherical(X, Y, Z):
+    r = np.sqrt(X ** 2 + Y ** 2 + Z ** 2)
+    theta = np.arctan2(np.sqrt(X ** 2 + Y ** 2), Z)
+    phi = np.arctan2(Y, X)
+    return (r, phi, theta)
+
+
+def cylindrical_to_cartesian(R, phi, z):
+    X = R * np.cos(phi)
+    Y = R * np.sin(phi)
+    Z = z
+    return (X, Y, Z)
+
+
+def cylindrical_to_spherical(R, phi, z):
+    r = np.sqrt(R ** 2 + z ** 2)
+    theta = np.arctan2(R, z)
+    phi = phi
+    return (r, phi, theta)
+
+
+def spherical_to_cartesian(r, theta, phi):
+    X = r * np.sin(theta) * np.cos(phi)
+    Y = r * np.sin(theta) * np.sin(phi)
+    Z = r * np.cos(theta)
+    return (X, Z, Y)
+
+
+def spherical_to_cylindrical(r, theta, phi):
+    R = r * np.sin(theta)
+    phi = phi
+    z = r * np.cos(theta)
+    return (R, z, phi)
+
+
+def no_op(*args):
+    return args
+
+
+def meshgrid_from_plane(coord, k, l, DEFAULT):
+    lgrid, kgrid = np.meshgrid(coord[l - 1], coord[k - 1])
+    tot = {1, 2, 3}
+    m = list(tot ^ {k, l})[0]
+    mgrid = DEFAULT[m - 1]
+    return [kgrid, lgrid, mgrid]
```

Comment:
I don't see a good reason to return a list (mutable) rather than a tuple (immutable)
```suggestion
    return kgrid, lgrid, mgrid
```